### PR TITLE
fix(nc-gui): search implemented for gallery view

### PR DESCRIPTION
**Pull request by reenphygeorge**
## Change Summary

Implemented search for gallery view

- [x] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

### CODE REVIEW

File: packages/nc-gui/composables/useViewData.ts [hunk 1]
```diff
@@ -160,14 +160,15 @@ export function useViewData(
   }
 
   async function loadData(params: Parameters<Api<any>['dbViewRow']['list']>[4] = {}) {
+    console.log(params)
```
Comment (wingkwong) [inline packages/nc-gui/composables/useViewData.ts:163]: please remove console log.

**Review by dstala — CHANGES_REQUESTED** [review]
- [ ] didn't work for non primary value field
- [ ] didn't work when connected to external db (tried with sakila PG)

**Review by wingkwong — CHANGES_REQUESTED** [review]
Your solution got a few problems. Suggest you read till the end first.

1. By hardcoding `(Title,like,%${search.value.query}%)`, you only assume the column is `Title`. Columns with other names would fail. Also, you cannot use `like` for bigint type. You may check out `xWhere` logic from either `useKanbanViewStore.ts` / `useSmartsheetStore.ts` for reference.

image 

2. After filtering, when you go to another views and switch back, the data should be filtered. In your case, filter text is here but the data is not filtered. This is because you only take care of `reloadViewDataHook` but not the view switch, i.e. the `loadData()` in `watch()`. 

Therefore, I think you may do the following

- undo everything you made
- retrieve `xWhere` from `useSmartsheetStoreOrThrow` in `Gallery.vue`, then pass `xWhere` to `useViewData` as the third parameter.

**Comment by wingkwong** [comment]
Moreover, I think the search logic got another problem. @dstala please help confirm. probably we may create a new issue for it.

Assuming there are two columns, `Title1` and `Title2`

- setting search as `Title2=123` in View A, the data is shown as expected.
- switch to View B
- switch back to View A
- search become `Title1=123`, the data is shown unexpectedly.

The expected behaviour should be still showing data with `Title2=123`.

File: packages/nc-gui/composables/useViewData.ts [hunk 2]
```diff
@@ -35,6 +35,7 @@ export function useViewData(
   meta: Ref<TableType | undefined> | ComputedRef<TableType | undefined>,
   viewMeta: Ref<ViewType | undefined> | ComputedRef<(ViewType & { id: string }) | undefined>,
   where?: ComputedRef<string | undefined>,
+  xWhere?:  ComputedRef<string | undefined>,
```
Comment (wingkwong) [inline packages/nc-gui/composables/useViewData.ts:38]: Here `xWhere` is useless. you don't need to modify this file at all. In fact, you are passing `xWhere` in `useViewData(meta, view, xWhere)` as the third parameter, which is `where` here.

**Review by wingkwong — CHANGES_REQUESTED** [review]

**Review by dstala — APPROVED** [review]

**Review by wingkwong — APPROVED** [review]

**Comment by dstala** [comment]
> Moreover, I think the search logic got another problem. @dstala please help confirm. probably we may create a new issue for it.
> 
> Assuming there are two columns, `Title1` and `Title2`
> 
> * setting search as `Title2=123` in View A, the data is shown as expected.
> * switch to View B
> * switch back to View A
> * search become `Title1=123`, the data is shown unexpectedly.
> 
> The expected behaviour should be still showing data with `Title2=123`.

@wingkwong created issue. Also asked Shreyas to redesign search box to include column name info display

**Review by dstala — APPROVED** [review]

**Pull request merged**